front/src/modules/simulationResult/components/ChartHelpers/enableInteractivity.tsx

```diff
@@ -0,0 +1,384 @@
+      .attr('y1', (d) => newY(d as number))
+      .attr('y2', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-text')
+      .attr('y', (d) => newY(d as number));
+  } else {
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-line')
+      .attr('y1', (d) => newY(d as number))
+      .attr('y2', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#get-operationalPointsZone .op-text')
+      .attr('y', (d) => newY(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-line')
+      .attr('x1', (d) => newX(d as number))
+      .attr('x2', (d) => newX(d as number));
+    chart.drawZone
+      .selectAll('#gev-operationalPointsZone .op-text')
+      .attr('x', (d) => newX(d as number));
+  }
+
+  chart.drawZone.selectAll('.curve-label').attr('transform', event.transform.toString());
+
+  // CLARA: removed conflictsPoints as it seems to never be created
+  return { newX, newY };
+};
+
+// Factorizes func to update VerticalLine on 3 charts: SpaceTime, SpeedSpaceChart, SpaceCurvesSlopes
+export const traceVerticalLine = (
+  chart: Chart | undefined,
+  keyValues: ChartAxes,
+  positionValues: PositionsSpeedTimes<Date>,
+  rotate: boolean,
+  timePosition: Date
+) => {
+  if (chart !== undefined) {
+    displayGuide(chart, 1);
+    if (rotate) {
+      chart.svg
+        .selectAll('#horizontal-line')
+        .attr(
+          'y1',
+          chart.y(
+            !isSpaceTimeChart(keyValues) && positionValues.speed
+              ? positionValues.speed.position
+              : timePosition
+          )
+        )
+        .attr(
+          'y2',
+          chart.y(
+            !isSpaceTimeChart(keyValues) && positionValues.speed
+              ? positionValues.speed.position
+              : timePosition
+          )
+        );
+    } else {
+      chart.svg
+        .selectAll('#vertical-line')
+        .attr(
+          'x1',
+          chart.x(
+            !isSpaceTimeChart(keyValues) && positionValues.speed
+              ? positionValues.speed.position
+              : timePosition
+          )
+        )
+        .attr(
+          'x2',
+          chart.x(
+            !isSpaceTimeChart(keyValues) && positionValues.speed
+              ? positionValues.speed.position
+              : timePosition
+          )
+        );
+    }
+
+    updatePointers(chart, keyValues, positionValues, rotate);
+  }
+};
+
+// enableInteractivity
+
+// Override the default wheelDelta computation to get smoother zoom
+function wheelDelta(event: WheelEvent) {
+  let factor = 1;
+  if (event.deltaMode === 1) {
+    factor = 0.005;
+  } else if (event.deltaMode) {
+    factor = 0.01;
+  } else {
+    factor = 0.002;
+  }
+
+  return -event.deltaY * factor;
+}
+
+export const enableInteractivity = <
+  T extends Chart | SpeedSpaceChart,
+  Data extends SimulationTrain<Date> | GevPreparedData | SpaceCurvesSlopesData
+>(
+  chart: T | undefined,
```

Comment:
Same here, it would be best to do this test in the component rather than in this function

Reply:
we can discuss that further if you want :)